### ISSUE TITLE
It would fail if parent is a template class

### ISSUE THREAD

**Issue by ryzzn** [mosra/m.css#35]
https://github.com/mosra/m.css/blob/eb2ee6f27e42e002a912c68c8815b5b5fa59db15/doxygen/dox2html5.py#L1750

Hi, 
Prefix won't contain parent's template parameters list but compound.name does contain, so the assertion would fail.

**Comment by mosra** [comment]
Hi, thanks for the report!

Can you provide a minimal repro case for this? Would make it much easier for me to fix and test. I guess it happens with nested classes or something? 

Thank you :)

**Comment by mosra** [comment]
Hi,

sorry, I have trouble reproducing this. Can you provide a minimal code snippet that triggers this assert? Thank you!

**Comment by ryzzn** [comment]
I'm sorry I haven't described the issue clearly.

Here's the code:
```c++
template <class T>
struct S
{
};

template <class T>
struct S<T*>
{
  struct SS {};
};
```

`S<T*>` is partial template specialization and failure message is:

```
 Traceback (most recent call last):
  File "./dox2html5.py", line 2897, in <module>
    run(doxyfile, os.path.abspath(args.templates), args.wildcard, args.index_pages, search_merge_subtrees=not args.search_no_subtree_merging, search_add_lookahead_barriers=not args.search_no_lookahead_barriers, search_merge_prefixes=not args.search_no_prefix_merging)
  File "./dox2html5.py", line 2796, in run
    postprocess_state(state)
  File "./dox2html5.py", line 1752, in postprocess_state
    assert compound.name.startswith(prefix)
AssertionError
```

Then I print the `compound` and `prefix` variables
```
namespace(brief='', children=[], has_details=<Element 'detaileddescription' at 0x7f0a9cf61048>, id='structS_3_01T_01_5_01_4_1_1SS', is_deprecated=False, kind='struct', name='S&lt;T*&gt;::SS', parent='structS', templates=None, url='structS_3_01T_01_5_01_4_1_1SS.html')

S::
```

**Comment by mosra** [comment]
Ahh, great, partial template specialization, that's what I missed. :)

As crazy as it might sound, it's not a problem in m.css but in fact related to a Doxygen bug I reported back in December: https://bugzilla.gnome.org/show_bug.cgi?id=791942 (a bug that appears to be completely unrelated).  As noted in the bug description, I "worked around" that by reverting the particular offending commit in my Doxygen fork (see https://github.com/mosra/doxygen/commit/5ac9b11435741c88c500c700a647c10ee1b4d296), but that unfortunately brings back some other problems (but minor problems, compared to this).

With that commit reverted, the `prefix` is correctly `S<T*>::` instead of just `S::` and everything is great. If you can apply the revert on your side, awesome, if not, I have an (ugly) workaround for you:

The problem boils down to the `*` and `&` characters, among others, so if the specialization contains them, it will fail. If you write the following, for example, it will work:

```cpp
template <class T>
struct S<typename std::add_pointer<T>::type>
{
  struct SS {};
};
```

Yes, *I know*, it's **really ugly**, but can do the job. Unfortunately this is not possible to fix in m.css, it's a problem in Doxygen itself and all I can do is just consume its output. But I'll push on that bug report, asking when it can be fixed :)

**Comment by mosra** [comment]
Added [a comment on the Doxygen issue](https://bugzilla.gnome.org/show_bug.cgi?id=791942#c1).

**Comment by ryzzn** [comment]
T_T, Thanks for reasoning this in detail. 

Finally I'll choose revert that commit to make work, because my project isn't small enough to change all code relating to partial template specialization. But I haven't tried it. If there is any result, I'll reply to you. Thanks so much.

**Comment by mosra** [comment]
Let's keep this open until the Doxygen issue is fixed. It causes numerous problems to me as well. There was some activity on it yesterday, so maybe this bug gets some attention too.

I would attempt to fix that myself, but this particular code is totally impenetrable for me and I gave up on trying to understand what is happening in it. Also with the lack of tests there I fear I would break other things by accident :/

**Comment by mosra** [comment]
For reference, doxygen/doxygen#6299 is the related Doxygen issue on the GitHub bugtracker, still no progress there :/

**Comment by mosra** [comment]
Fix for this issue is in https://github.com/doxygen/doxygen/commit/449a7e2b4ff114a72be573013558bae19672ebbc, scheduled to be released as 1.8.15 most probably at the end of the year.

**Comment by mosra** [comment]
Doxygen 1.8.15 got released half a year ago, not sure why is this still open :)
